frontend/components/atoms/buttons/button.php

```diff
@@ -1,35 +1,21 @@
 <?php
 
-$class  = $args['class'] ?? '';
-$label  = $args['label'] ?? 'press me';
-$url    = $args['url'] ?? '';
-$target = $args['target'] ?? '';
+switch ( $args['button_style'] ?? 'default' ) {
+	case 'link':
+		$class = 'text-regular text-link-primary cursor-pointer underline hover:no-underline inline-block';
+		break;
+	default:
+		$class = 'text-regular text-primary-btn bg-primary-btn-bg text-link-primary cursor-pointer px-primary-btn-padding-x py-primary-btn-padding-y hover:opacity-90 inline-block';
+		break;
+}
 
-$target = empty( $target ) ? '_self' : $target;
-
-$class .= ' inline-block py-2 px-2 font-extrabold text-primary border-2 transition-2s ';
 ?>
 
-<?php if ( $url ) : ?>
-
-	<a
-		data-atom="buttons/button"
-		class="<?php echo esc_attr( $class ); ?>"
-		href="<?php echo esc_url( $url ); ?>"
-		target="<?php echo esc_attr( $target ); ?>">
-
-		<?php echo esc_html( $label ); ?>
-
-	</a>
-
-<?php else : ?>
-
-	<button
-		data-atom="buttons/button"
-		class="<?php echo esc_attr( $class ); ?>">
-
-		<?php echo esc_html( $label ); ?>
-
-	</button>
+<a
+	data-type="atom/buttons/button"
+	class="<?php echo esc_attr( $class ); ?>"
+	href="<?php echo esc_url( $args['link']['url'] ?? '#' ); ?>"
```

Comment:
components should receive the data without an outer context format.
Instead of $args['link']['url'] 
should be $args['url']

Reply:
I did it this way so it matches the format that comes out of the ACF link field. I didn't see much value in restructuring the data before passing it in.

Reply:
ACF is not the only context where a component's data can come from

Reply:
Agreed, but it's the most common so I think we should follow this standard as much as possible to save additional work.

Reply:
hmm the additional work is not much. Here we depend on a 3rd party plugin that can change in a future version.
the point is that a component should be totally independant, isolated from the data outer context and we are encouraging a good practice here. This will help to reuse the component in other context or projects. 

please read the section Build and Test in isolation from
https://blog.bitsrc.io/atomic-design-and-ui-components-theory-to-practice-f200db337c24

Reply:
Thanks for sharing. See my comment below, I do agree that components should be independent but I don't think this affects it.